virtio-devices/src/console.rs

```diff
@@ -332,6 +473,46 @@ impl EpollHelperHandler for ConsoleEpollHandler {
         }
         Ok(())
     }
+
+    fn handle_timeout(&mut self, helper: &mut EpollHelper) -> Result<(), EpollHelperError> {
+        if !self.endpoint.is_pty() {
+            return Ok(());
+        }
+
+        if self.file_event_registered {
+            // This very specific case happens when the console is connected
+            // to a PTY. We know EPOLLHUP is always present when there's nothing
+            // connected at the other end of the PTY. That's why getting no event
+            // means we can flush the output of the console through the PTY.
+            self.trigger_pty_flush()
+                .map_err(EpollHelperError::HandleTimeout)?;
+        }
+
+        self.register_file_event(helper)
+    }
+
+    fn event_list(
```

Comment:
It would be good to have some comments to explain what `event_list()` does. My understanding is that it is for flushing buffered pty output as soon as possible, e.g. do not wait until we have the next `EPOLLIN` 'file event'.

Reply:
I've added a comment.